wordpress-mobile/WordPress-iOS
# Domain Management: Implement add new freestanding domain flow

Closes #21778 
Closes #21780 
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/644

## Description
This PR adds the first steps for adding a domain which is searching for a new domain. It also integrates with the already implemented "Choice" screen and hooks up the freestanding domain purchase flow.

## Recording

https://github.com/wordpress-mobile/WordPress-iOS/assets/25306722/1be653e8-17aa-4c33-a15b-f164eab7724b

## Testing Instructions

1. Setup the store sandbox
2. Navigate to the "Me" tab
3. Tap on "My Domains"
4. Tap on the + button
5. ✅ Make sure that the search domains screen is displayed as a modal
6. Search for a domain
7. ✅ Make sure that the search works fine
8. Select a .live domain
9. Tap on "Select Domain"
10. ✅ Make sure the purchase choice screen is displayed
11. Tap on "Get Domain"
12. ✅ Make sure the checkout page is displayed (P.S: There is a delay here, and ideally we should display an activity indicator. Check #21978 )
13. Go through the checkout flow and purchase the domain
14. ✅ Make sure the modal is dismissed after purchasing is successful and the domains list is updated

Other than the above flow, please testing the existing domain purchasing flows for regressions:
* My Site -> Domains
* Dashboard -> Free domain with plan card
* Site creation

## Regression Notes
1. Potential unintended areas of impact
Other places where a domain can be purchased:
2. What I did to test those areas of impact (or what existing automated tests I relied on)
Testing manually
3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
